src/phase0/state_transition/epoch_processing.rs

```diff
@@ -154,8 +156,48 @@ pub fn process_registry_updates<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    // Process activation eligibility and ejections
+    let current_epoch = get_current_epoch(state, context);
+
+    for i in 0..state.validators.len() {
+        let validator = &mut state.validators[i];
+        if is_eligible_for_activation_queue(validator, context) {
+            validator.activation_eligibility_epoch = current_epoch + 1;
+        }
+
+        if is_active_validator(validator, current_epoch)
+            && validator.effective_balance <= context.ejection_balance
+        {
+            initiate_validator_exit(state, i, context);
+        }
+    }
+
+    // Queue validators eligible for activation and not yet dequeued for activation
+    let mut activation_queue = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter(|&(_, validator)| is_eligible_for_activation(state, validator))
+        .map(|(i, _)| i)
+        .collect::<Vec<ValidatorIndex>>();
+    // Order by the sequence of activation_eligibility_epoch setting and then index
+    activation_queue.sort_by(|i, j| {
+        let a = &state.validators[*i];
+        let b = &state.validators[*j];
+        (a.activation_eligibility_epoch, i).cmp(&(b.activation_eligibility_epoch, j))
+    });
+
+    // Dequeued validators for activation up to churn limit
+    for i in activation_queue
+        .iter()
```

Comment:
i would actually do this and just take the indices by value
```suggestion
        .into_iter()
```

the idea here is that you will already be shuffling say 64 bits around, and taking them by reference mean you will need to deref the values to get the 64 bits you need